token/services/db/sql/common/tokens.go

```diff
@@ -949,13 +980,68 @@ func (db *TokenDB) SetSupportedTokenFormats(formats []token.Format) error {
 	return nil
 }
 
-func (db *TokenDB) getSupportedTokenTypes() []token.Format {
+func (db *TokenDB) getSupportedTokenFormats() []token.Format {
 	db.sttMutex.RLock()
 	supportedTokenTypes := db.supportedTokenFormats
 	db.sttMutex.RUnlock()
 	return supportedTokenTypes
 }
 
+func (db *TokenDB) unspendableTokenFormats(ctx context.Context, walletID string, tokenType token.Type) ([]token.Format, error) {
+	span := trace.SpanFromContext(ctx)
+	where, args := common.Where(db.ci.HasTokenDetails(driver.QueryTokenDetailsParams{
+		WalletID:  walletID,
+		TokenType: tokenType,
+		Spendable: driver.Any,
+	}, ""))
+	query, err := NewSelectDistinct("ledger_type").From(db.table.Tokens).Where(where).Compile()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to compile query")
+	}
+	logger.Debug(query, args)
+	span.AddEvent("start_query", tracing.WithAttributes(tracing.String(QueryLabel, query)))
+	rows, err := db.db.Query(query, args...)
+	span.AddEvent("end_query")
+	if err != nil {
+		return nil, errors.Wrapf(err, "error querying db")
+	}
+	if err := rows.Err(); err != nil {
+		return nil, errors.Wrapf(err, "error querying db")
+	}
+	defer rows.Close()
+	// read the types from the query result and remove discard those in db.getSupportedTokenFormats()
+	supportedFormats := db.getSupportedTokenFormats()
+	logger.Debugf("supported token formats are [%v]", supportedFormats)
+	includeFormats := make([]token.Format, 0)
+	for rows.Next() {
+		var tmp string
+		if err := rows.Scan(&tmp); err != nil {
+			return nil, errors.Wrapf(err, "failed to scan row")
+		}
+
+		logger.Debugf("format from db [%s]", tmp)
+
+		// include tmp only if it is not in supportedFormats
+		format := token.Format(tmp)
+		found := false
+		for _, t := range supportedFormats {
```

Comment:
We can either use `slices.Contains(supportedFormats, token.Format(tmp))` or (in case we expected there to be many supported formats) use
```
supportedFormats collection.Set[token.Format]
```
and then `supportedFormats.Contains(token.Format(tmp))`

Reply:
okay